src/Logging.XUnit/XUnitLoggerProvider.cs

```diff
@@ -58,7 +26,20 @@ public XUnitLoggerProvider(ITestOutputHelperAccessor accessor, XUnitLoggerOption
         }
 
         /// <inheritdoc />
-        public virtual ILogger CreateLogger(string categoryName) => new XUnitLogger(categoryName, _accessor, _options);
+        public virtual ILogger CreateLogger(string categoryName)
+        {
+            if (_outputHelperAccessor != null)
+            {
+                return new XUnitLogger(categoryName, _outputHelperAccessor, _options);
+            }
+
+            if (_messageSinkAccessor != null)
+            {
+                return new XUnitLogger(categoryName, _messageSinkAccessor, _options);
+            }
+
+            throw new InvalidOperationException($"Either {nameof(_outputHelperAccessor)} or {nameof(_messageSinkAccessor)} must not be null.");
```

Comment:
Maybe invert the message to be _"you must set something"_ rather than _"you've not set either"_?

Reply:
Addressed in a752efb30e8988ef3c6cc1b12f89e8c12b29b4b7, see also https://github.com/martincostello/xunit-logging/pull/246#discussion_r720685050